chunk/chunk_store_test.go

```diff
@@ -146,3 +146,44 @@ func diff(want, have interface{}) string {
 	})
 	return "\n" + text
 }
+
+func TestBigBuckets(t *testing.T) {
+	scenarios := []struct {
+		from, through, dailyBucketsFrom model.Time
+		buckets                         []string
+	}{
+		{
+			from:             model.TimeFromUnix(0),
+			through:          model.TimeFromUnix(0).Add(3*24*time.Hour) - 1,
+			dailyBucketsFrom: model.TimeFromUnix(0).Add(1 * 24 * time.Hour),
+			buckets:          []string{"0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19", "20", "21", "22", "23", "24", "d1", "d2"},
+		},
+		{
+			from:             model.TimeFromUnix(0),
+			through:          model.TimeFromUnix(0).Add(3*24*time.Hour) - 1,
+			dailyBucketsFrom: model.TimeFromUnix(0).Add(2*24*time.Hour) - 1, // Only the day matters for the epoch start time, not the time.
+			buckets:          []string{"0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19", "20", "21", "22", "23", "24", "d1", "d2"},
+		},
+		{
+			from:             model.TimeFromUnix(0),
+			through:          model.TimeFromUnix(0).Add(3*24*time.Hour) - 1,
+			dailyBucketsFrom: model.TimeFromUnix(0).Add(1*24*time.Hour) - 1,
+			buckets:          []string{"0", "d0", "d1", "d2"},
+		},
+		{
+			from:             model.TimeFromUnix(0),
+			through:          model.TimeFromUnix(0).Add(2 * 24 * time.Hour),
+			dailyBucketsFrom: model.TimeFromUnix(0).Add(99 * 24 * time.Hour),
+			buckets:          []string{"0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19", "20", "21", "22", "23", "24", "25", "26", "27", "28", "29", "30", "31", "32", "33", "34", "35", "36", "37", "38", "39", "40", "41", "42", "43", "44", "45", "46", "47", "48"},
+		},
```

Comment:
Maybe add another case that checks when `dailyBucketsFrom` is _before_ the interval?